### DISCUSSION TITLE
How to get the texture offset and repeat values?

### DISCUSSION THREAD

**Discussion by CITIZENDOT** [donmccurdy/glTF-Transform#1094] [Q&A] [answered]
three.js has [Texture.offset](https://threejs.org/docs/#api/en/textures/Texture.offset) and [Texture.repeat](https://threejs.org/docs/#api/en/textures/Texture.repeat). I know this is not included in GLTF spec, I just want to know if there is a way to access these properties.

**Comment by donmccurdy** [comment, marked as answer]
In glTF this is handled by the `KHR_texture_transform` extension. You can see glTF Transform's implementation here:

- https://gltf-transform.dev/modules/extensions/classes/KHRTextureTransform
- https://gltf-transform.dev/modules/core/classes/TextureInfo

To get the offset and scale of the base color texture, for example, you'd write:

```js
const textureInfo = material.getBaseColorTextureInfo();
const transform = textureInfo.getExtension('KHR_texture_transform');

if (transform) {
  const offset = transform.getOffset();
  const rotation = transform.getRotation();
  const scale = transform.getScale(); // equivalent to THREE.Texture's .repeat property
}
```

**Reply by CITIZENDOT** [reply]
That worked. Thank you.
